sql/catalyst/src/main/scala/org/apache/spark/sql/types/StructType.scala

```diff
@@ -511,6 +511,30 @@ case class StructType(fields: Array[StructField]) extends DataType with Seq[Stru
   @transient
   private[sql] lazy val interpretedOrdering =
     InterpretedOrdering.forSchema(this.fields.map(_.dataType))
+
+  /**
+   * Parses the text representing constant-folded default column literal values.
+   * @return a sequence of either (1) NULL, if the column had no default value, or (2) an object of
+   *         Any type suitable for assigning into a row using the InternalRow.update method.
+   */
+  private [sql] lazy val defaultValues: Array[Any] =
```

Comment:
Shall we rename this as existenceDefaultValues instead?

Reply:
Thanks Gengliang, that's a good idea, I can rename that in the next PR :)

Reply:
Is this because default values do not exist sometimes, or is it because `default` value can mean something like [`DecimalType.SYSTEM_DEFAULT`](https://github.com/apache/spark/blob/58d3f1516ed812b692709991e551829aa0090578/sql/catalyst/src/main/scala/org/apache/spark/sql/types/DecimalType.scala#L129) or `0` for non-nullable numeric types? 

`definedDefaultValues` as is sounds sane to me. Maybe we can rename something like `userDefinedDefaultValues` or `userSpecifiedDefaultValues` if there's the concern.

BTW, `existenceDefaultValues` would have to be `existentDefaultValues` or `existingDefaultValues`? I am okay with this. Don't feel strongly. Just thinking load :-).

Reply:
I added a comment about this to the class field in the next PR.